aggregate_root/spec/aggregate_root_spec.rb

```diff
@@ -10,12 +10,12 @@
     expect(order).to receive(:apply_order_created).with(order_created).and_call_original
     order.apply(order_created)
     expect(order.status).to eq :created
-    expect(order.expected_events).to eq([order_created])
+    expect(order.unpublished_events.to_a).to eq([order_created])
```

Comment:
it seems to be annoying that (tests show that here), that we need to add `to_a` always with the solution suggested.

Reply:
You can use `each`, `map` without adding `to_a`. You only need `to_a` when you want an Array which makes sense to me.

Reply:
I think that name is messing up with expectation, looking on plural form you are expecting to have arrays or array like ;)

Reply:
And an iterator is array-like.